ui/raidboss/data/00-misc/test.ts

```diff
@@ -356,10 +356,28 @@ const triggerSet: TriggerSet<Data> = {
         },
       },
     },
+    {
+      id: 'Test OutputStrings',
+      type: 'GameLog',
+      netRegex: NetRegexes.echo({ line: 'cactbot test outputStrings', capture: false }),
+      infoText: (data, _matches, output) => {
+        const obj = {
+          ...data,
+          toString: () => data.me,
+        };
+        return output.text!({ customData: obj });
+      },
+      outputStrings: {
+        text: {
+          en: 'data = ${customData}, data.job = ${customData.job}',
```

Comment:
This PR is definitely along similar thoughts I've had in the past and I am super positive on this approach.  Thanks for taking the time to do this; I had no idea this was on your radar.  (I run my own static, so don't ever think about people as "roles" or "jobs" and know everybody by name.  I realize this is probably not at all how a lot of other folks play, but it's why things are as they are.)

I think that the follow-up here would be to add some option for default way to display players (defaulting to nicknames to be consistent with existing code) and then other names, and then we can replace calls to `ShortName` with some new "create a player object" function and pass them to output strings.

One question I do have is: I think I agree with this general approach, in terms of providing an object so that job/role/name/nick etc can all be used by a trigger.  However, are there other uses past players? Mostly just trying to understand what other uses cases might look like or if it's really just "players".

Other questions are: I think we have a lot of testing right now to make sure that the loose coupling between calls to `output` and entries in `outputStrings` all match up.  Is there anything we can do here for testing "sub params" or is the general thought that cactbot would not use them, and selection of sub params would be done via options or user overrides most of the time?

I do think some more investigation needs to happen on https://github.com/OverlayPlugin/OverlayPlugin/issues/60 though.  Cactbot doesn't use a ton of job/role triggers and there's definitely periodic issues with the party info being incorrect.  If they start being optional for all triggers then it will start happening more.

Reply:
> I realize this is probably not at all how a lot of other folks play, but it's why things are as they are.

I think the biggest case for role or job-based callouts are for party finder groups. That's been the case every time something like this was requested in Discord, at least.

> I think that the follow-up here would be to add some option for default way to display players (defaulting to nicknames to be consistent with existing code) and then other names, and then we can replace calls to `ShortName` with some new "create a player object" function and pass them to output strings.

A default config option is intended, when this is more fully fleshed out.

> One question I do have is: I think I agree with this general approach, in terms of providing an object so that job/role/name/nick etc can all be used by a trigger. However, are there other uses past players? Mostly just trying to understand what other uses cases might look like or if it's really just "players".

I can't think of any other use cases currently, but I deliberately wrote this preliminary code to not tie in to any sort of explicit structure so that it can support any object type.

> Other questions are: I think we have a lot of testing right now to make sure that the loose coupling between calls to `output` and entries in `outputStrings` all match up. Is there anything we can do here for testing "sub params" or is the general thought that cactbot would not use them, and selection of sub params would be done via options or user overrides most of the time?

It's probably possible, but I haven't looked too far into the testing code in general. Tweaks to the test code in this PR were only to allow the new object structure to pass through, for example, and should be tightened in some manner.

> I do think some more investigation needs to happen on [OverlayPlugin/OverlayPlugin#60](https://github.com/OverlayPlugin/OverlayPlugin/issues/60) though. Cactbot doesn't use a ton of job/role triggers and there's definitely periodic issues with the party info being incorrect. If they start being optional for all triggers then it will start happening more.

I have some thoughts on how to handle this issue better, but debugging it and testing fixes is a pain because of how inconsistent the issue is, which puts it further down my list of things I'd like to try and fix myself.

Reply:
> I have some thoughts on how to handle this issue better, but debugging it and testing fixes is a pain because of how inconsistent the issue is, which puts it further down my list of things I'd like to try and fix myself.

If you need assistance being in a party and disconnecting / etc, I'm happy to come help be part of any debugging.